aquacomputer_d5next.c

```diff
@@ -1658,24 +1666,14 @@ static int aqc_write(struct device *dev, enum hwmon_sensor_types type, u32 attr,
 					return -EINVAL;
 				}
 
-				ctrl_mode += channel;
+				if (val > 0)
+					ctrl_mode += channel;
 
 				ctrl_values_offsets[0] = priv->fan_ctrl_offsets[channel] +
 				    AQUAERO_FAN_CTRL_SRC_OFFSET;
 				ctrl_values[0] = ctrl_mode;
 				ctrl_values_types[0] = AQC_BE16;
-
-				if (val == 0) {
```

Comment:
Setting pwm_enable to 0 should spin the fan to 100% and relinquish control of it that way. This real automatic mode can be activated using a higher value - but I first need to understand what it's all about, sorry if it's something obvious.

Reply:
As you said we want to not control the fan when setting `pwm_enable` to 0,but we currently fake this mode as described above. On Aquaero setting the internal control mode to `0xffff` signifies that no valid mode has been set and it defaults to 100% speed.

Reply:
Ah, that clears it up, thanks.